src/main/java/com/alextherapeutics/diga/DigaApiClient.java

```diff
@@ -248,7 +258,12 @@ private DigaCodeValidationResponse performCodeValidation(DigaCodeInformation cod
   private DigaInvoiceResponse performDigaInvoicing(
       DigaInvoice invoice, DigaBillingInformation billingInformation, DigaProcessCode processCode)
       throws DigaXmlWriterException {
-    var xmlInvoice = xmlRequestWriter.createBillingRequest(invoice, billingInformation);
+    var xmlInvoice =
+        invoice instanceof DigaCorrectionInvoice
+            ? xmlRequestWriter.createInvoiceCorrectionRequest(
+                (DigaCorrectionInvoice) invoice, billingInformation)
+            : xmlRequestWriter.createBillingRequest(invoice, billingInformation);
+
```

Comment:
with java 17 you can do this instead, which is nicer I think:

```java
var xmlInvoice =
        invoice instanceof DigaCorrectionInvoice correctionInvoice
            ? xmlRequestWriter.createInvoiceCorrectionRequest(
                correctionInvoice, billingInformation)
            : xmlRequestWriter.createBillingRequest(invoice, billingInformation);
```

then you don't get that ugly-looking cast

it's really only syntactic sugar though, since you already checked the instance there shouldn't be any issues with casting here. But maybe you'd like to make this change anyway?

Reply:
That's great. Changed